src/osdc/Objecter.cc

```diff
@@ -3435,6 +3435,7 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
     // new tid
     s->ops.erase(op->tid);
     op->tid = ++last_tid;
+    s->ops[op->tid] = op; // add op to ops again
```

Comment:
@gregsfortytwo The rbd case uses the localized read flag so it wouldn't hit that block. Of course, I'd imagine that would just result in the op ping-ponging back and forth to the replica in the case where it's being recovered. Perhaps that block should also include localized reads?